src/dev/code_coverage/ingest_coverage/transforms.js

```diff
@@ -83,20 +87,53 @@ export const staticSite = (urlBase) => (obj) => {
   return { ...obj, staticSiteUrl: prokForBoth() };
 };
 
+const leadingSlashRe = /^\//;
+export const maybeDropLeadingSlash = (x) =>
+  leadingSlashRe.test(x) ? Either.right(x) : Either.left(x);
+export const dropLeadingSlash = (x) => x.replace(leadingSlashRe, '');
+export const stripLeading = (x) => maybeDropLeadingSlash(x).fold(id, dropLeadingSlash);
+
 export const coveredFilePath = (obj) => {
   const { staticSiteUrl, COVERAGE_INGESTION_KIBANA_ROOT } = obj;
 
   const withoutCoveredFilePath = always(obj);
-  const leadingSlashRe = /^\//;
-  const maybeDropLeadingSlash = (x) => (leadingSlashRe.test(x) ? Either.right(x) : Either.left(x));
-  const dropLeadingSlash = (x) => x.replace(leadingSlashRe, '');
-  const dropRoot = (root) => (x) =>
-    maybeDropLeadingSlash(x.replace(root, '')).fold(id, dropLeadingSlash);
+  const dropRoot = (root) => (x) => stripLeading(x.replace(root, ''));
   return maybeTotal(staticSiteUrl)
     .map(dropRoot(COVERAGE_INGESTION_KIBANA_ROOT))
     .fold(withoutCoveredFilePath, (coveredFilePath) => ({ ...obj, coveredFilePath }));
 };
 
+const findTeam = (x) => x.match(/.+\s{1,3}(.+)$/, 'gm');
+export const pluckIndex = (idx) => (xs) => xs[idx];
+const pluckTeam = pluckIndex(1);
+
+export const teamAssignment = (teamAssignmentsPath) => (log) => async (obj) => {
```

Comment:
is there any benefit over defining function as `async teamAssignment(teamAssignmentsPath, log, obj)` similar to `assignTeam` with last argument `obj` being optional?

Reply:
actually, I'm not sure that should be async lol

Reply:
ok, now that I checked that.
I'm prolly not following what you are getting at so I'll just explain a little, maybe that helps.

So, `assignTeam` is a _private_ fn, while `teamAssignment` is exported.

Reply:
Ok, so back to function expression question, usually I put large objects last. 
Here, it's not as useful, until I start composing lazy functions.
In a fn composition, you do not want to hold onto memory objects, that are expensive.  
Instead, you pass them through, in one location...
E.g.: https://github.com/wayneseymour/kibana/blob/6d0e2464c454f77bad1a57b0111797de822e110a/src/dev/code_coverage/ingest_coverage/team_assignment/parse_owners.js#L27

^^ On line 32, this composition is used, and the arg that `dropCCDelim` & `pathAndTeams` is passed through (point free).  So, imagine that the entire transformation could be one composition (coming soon, using transducers [particularly if speed becomes an issue])

Does that help?  We can always hop on zoom if that's more helpful.